nullaway/src/main/java/com/uber/nullaway/fixserialization/Serializer.java

```diff
@@ -164,4 +165,32 @@ private void appendToFile(String row, Path path) {
       throw new RuntimeException("Error happened for writing at file: " + path, e);
     }
   }
+
+  /**
+   * Converts the given uri to the real path. Note, in NullAway CI tests, source files exists in
+   * memory and there is no real path leading to those files. Instead, we just serialize the path
+   * from uri as the full paths are not checked in tests.
+   *
+   * @param uri Given uri.
+   * @return Real path for the give uri.
+   */
+  @Nullable
+  public static Path pathToSourceFileFromURI(@Nullable URI uri) {
+    if (uri == null) {
+      return null;
+    }
+    if ("jimfs".equals(uri.getScheme())) {
+      return Paths.get(uri);
+    }
+    if (!"file".equals(uri.getScheme())) {
+      return null;
+    }
+    Path path = Paths.get(uri);
+    try {
+      return path.toRealPath();
+    } catch (IOException e) {
+      // This is expected to happen in tests, we should just return the path from uri.
+      return path;
```

Comment:
Does this still apply?  Or is the test case handled in the `"jimfs"` case above?

Reply:
Yes, that is covered, but I still think it might be good to have a serialization instead of returning `null` and not serializing anything. Please let me know if you would like me to remove it.

Reply:
That's ok with me.  Let's just update the code comment